tests/ops/qutrit/test_qutrit_parametric_ops.py

```diff
@@ -0,0 +1,358 @@
+        res2 = dot_broadcasted(op_d.matrix(), op.matrix())
+        I = [np.eye(3 ** len(op.wires))] * op.batch_size
+        assert np.allclose(res1, I, atol=tol)
+        assert np.allclose(res2, I, atol=tol)
+        assert op.wires == op_d.wires
+
+
+class TestParameterFrequencies:
+    @pytest.mark.parametrize("op", PARAMETRIZED_OPERATIONS)
+    def test_parameter_frequencies_match_generator(self, op, tol):
+        """Check that parameter frequencies of parametrized operations are defined correctly."""
+        if not qml.operation.has_gen(op):
+            pytest.skip(f"Operation {op.name} does not have a generator defined to test against.")
+
+        gen = op.generator()
+
+        try:
+            mat = gen.matrix()
+        except (AttributeError, qml.operation.MatrixUndefinedError):
+
+            if isinstance(gen, qml.Hamiltonian):
+                mat = qml.utils.sparse_hamiltonian(gen, level=3).toarray()
+            elif isinstance(gen, qml.SparseHamiltonian):
+                mat = gen.sparse_matrix().toarray()
+            else:
+                pytest.skip(f"Operation {op.name}'s generator does not define a matrix.")
+
+        gen_eigvals = np.round(np.linalg.eigvalsh(mat), 8)
+        freqs_from_gen = qml.gradients.eigvals_to_frequencies(tuple(gen_eigvals))
+
+        freqs = op.parameter_frequencies
+        assert np.allclose(freqs, freqs_from_gen, atol=tol)
+
+
+# TODO: Add tests for decompositions
+
+
+matrix_data = [
+    (qml.TRX, 0, [0, 1], np.eye(3)),
+    (qml.TRX, 0, [1, 2], np.eye(3)),
+    (qml.TRX, 0, [0, 2], np.eye(3)),
+    (
+        qml.TRX,
+        np.pi / 2,
+        [0, 1],
+        np.array([[1, -1j, 0], [-1j, 1, 0], [0, 0, np.sqrt(2)]]) / np.sqrt(2),
+    ),
+    (
+        qml.TRX,
+        np.pi / 2,
+        [1, 2],
+        np.array([[np.sqrt(2), 0, 0], [0, 1, -1j], [0, -1j, 1]]) / np.sqrt(2),
+    ),
+    (
+        qml.TRX,
+        np.pi / 2,
+        [0, 2],
+        np.array([[1, 0, -1j], [0, np.sqrt(2), 0], [-1j, 0, 1]]) / np.sqrt(2),
+    ),
+    (qml.TRX, np.pi, [0, 1], -1j * np.array([[0, 1, 0], [1, 0, 0], [0, 0, 1j]])),
+    (qml.TRX, np.pi, [1, 2], -1j * np.array([[1j, 0, 0], [0, 0, 1], [0, 1, 0]])),
+    (qml.TRX, np.pi, [0, 2], -1j * np.array([[0, 0, 1], [0, 1j, 0], [1, 0, 0]])),
+    (
+        qml.TRX,
+        np.array([np.pi / 2] * 2),
+        [0, 1],
+        np.tensordot(
+            [1, 1], np.array([[1, -1j, 0], [-1j, 1, 0], [0, 0, np.sqrt(2)]]) / np.sqrt(2), axes=0
+        ),
+    ),
+    (
+        qml.TRX,
+        np.array([np.pi / 2] * 2),
+        [1, 2],
+        np.tensordot(
+            [1, 1], np.array([[np.sqrt(2), 0, 0], [0, 1, -1j], [0, -1j, 1]]) / np.sqrt(2), axes=0
+        ),
+    ),
+    (
+        qml.TRX,
+        np.array([np.pi / 2] * 2),
+        [0, 2],
+        np.tensordot(
+            [1, 1], np.array([[1, 0, -1j], [0, np.sqrt(2), 0], [-1j, 0, 1]]) / np.sqrt(2), axes=0
+        ),
+    ),
+]
+
+
+matrix_data_tf = [
```

Comment:
Test replacing this with the regular matrix data for the tensorflow tests and remove depending on output.